File: aurora/api/planning_api.py
# ======================================================================
# FILE: aurora/api/planning_api.py 
# START: PLANNING_API_COMPATIBILITY_FACADE
# ======================================================================
"""Compatibility façade for the Planning subsystem API."""

from aurora.subsystems.planning.api import planning_endpoint

__all__ = ["planning_endpoint"]
# ======================================================================
# END: PLANNING_API_COMPATIBILITY_FACADE 
# ======================================================================

File: aurora/static/aurora/js/planning/api/planning_api.js
// ======================================================================
// FILE: aurora/static/aurora/js/planning/api/planning_api.js
// START: PLANNING_API_CLIENT
// ======================================================================
(function(window, $) {
    "use strict";

    const Planning = window.AuroraPlanning = (
        window.AuroraPlanning || {}
    );

    Planning.api = Planning.api || {};

    function fetchPlanningData(endpoint, requestData) {
        return $.ajax({
            url: endpoint,
            method: "GET",
            dataType: "json",
            data: requestData || {},
        });
    }

    Planning.api.planning = {
        fetch: fetchPlanningData,
    };
})(window, jQuery);
// ======================================================================
// END: PLANNING_API_CLIENT
// ======================================================================